rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/resources/tests/NodeDefaultFilterDropdown.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { mount, VueWrapper } from "@vue/test-utils";
+import NodeDefaultFilterDropdown from "../NodeDefaultFilterDropdown.vue";
+import mitt, { Emitter, EventType } from "mitt";
+
+jest.mock("@/library/rundeckService", () => {
+  return {
+    getRundeckContext: jest.fn().mockReturnValue({
+      rdBase: "mockRdBase",
+      projectName: "test-project",
+      eventBus: {
+        on: jest.fn(),
+        emit: jest.fn(),
+        off: jest.fn(),
+      },
+    }),
+  };
+});
+
+const mockNodeSummary = {
+  filters: [
+    { filterName: "filter1", filter: "filter1" },
+    { filterName: "filter2", filter: "filter2" },
+  ],
+  totalCount: 2,
+};
+
+const mountNodeDefaultFilterDropdown = async (
+  props = {},
+  options: Record<string, any> = {},
+) => {
+  const wrapper = mount(NodeDefaultFilterDropdown, {
+    props: {
+      isDefaultFilter: false,
+      nodeSummary: mockNodeSummary,
+      ...props,
+    },
+    global: {
+      stubs: {
+        dropdown: {
+          template: '<div><slot></slot><slot name="dropdown"></slot></div>',
+        },
+        btn: true,
+      },
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+  await wrapper.vm.$nextTick();
+  return wrapper as VueWrapper<InstanceType<typeof NodeDefaultFilterDropdown>>;
+};
+describe("NodeDefaultFilterDropdown Component", () => {
+  let eventBus: Emitter<Record<EventType, any>>;
+  beforeAll(() => {
+    eventBus = mitt();
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it("triggers filter event when a saved filter is selected", async () => {
+    const wrapper = await mountNodeDefaultFilterDropdown();
+    const filterLink = wrapper.find('a[data-testid="saved-filter"]');
+    await filterLink.trigger("click");
+    expect(wrapper.emitted().filter[0]).toEqual([
+      { filterName: "filter1", filter: "filter1" },
+    ]);
+  });
+  it("sets a filter as default", async () => {
+    const emitSpy = jest.spyOn(eventBus, "emit");
+    const wrapper = await mountNodeDefaultFilterDropdown();
+    eventBus.emit("nodefilter:action:setDefault", "filter1");
+    const setDefaultLink = wrapper.find(
+      'a[data-testid="set-default-filter-link"]',
+    );
+    expect(setDefaultLink.exists()).toBe(true);
+    await setDefaultLink.trigger("click");
+    expect(emitSpy).toHaveBeenCalledWith(
+      "nodefilter:action:setDefault",
+      "filter1",
+    );
+  });
```

Comment:
Blocker: Few things to adjust in this test and possibly in the mocks:

1. Mitt should be used when there's a need for an external eventBus to trigger an event that the component being tested reacts to.  This is defined by `this.eventBus.on([eventName], methodToExecuteOnceEventHappens)` ie:

```
 this.eventBus.on("plugin-configs-saved", (configs: any) =>
      this.refreshAll(),
    );
```

So the first thing to do is to remove mitt and replace it with the usual mock for the eventBus, as NodeDefaultFilterDropdown only emits events and doesn't react to them;

2. Line 71 should be removed;

3. Adding context before point 3: When a filter is set to default, its value is saved to the NodeSummary with the key `defaultFilter`, and the NodeCard component which is the parent of NodeDefaultFilterDropdown, will figure out if the selected filter is equal to `.*` and update the flag being passed to the prop `isDefaultFilter` (true if it's equal to ".*")

There's a decision to make here, that is if this test will start with a default filter or not. Both scenarios can happen, but then:

- option 1: If the decision is to start with a default filter, then nodeSummary mock needs to be updated to add the key `defaultFilter` with a string value (ideally with one of the filterNames in the mock, or ".*" - remembering that if the later, then `isDefaultFilter` needs to be set to true); test should assert what was rendered before and after the click happens, and the mock receives another value for `defaultFilter` as well (defaultFilter can become null if the test click to remove the default filter);
  
- option 2: If the decision is to start without a default filter, then it's not necessary to tweak nodeSummary, but the rest should be updated the same: the test should assert what was rendered before and after the click happens,  and the mock receives a value for `defaultFilter` as well;
  
Please let me know if you need more context. The ideal solution would be to cover both options and in the same test, remove the default filter.